src/psm/src/ir_solver.cpp

```diff
@@ -705,8 +711,25 @@ bool IRSolver::CreateGmat(bool connection_only)
     vector<Node*>::iterator node_it;
     for (node_it = RDL_nodes.begin(); node_it != RDL_nodes.end(); ++node_it) {
       Node* node = *node_it;
-      m_C4Nodes.push_back(make_pair(node->GetGLoc(), v));
-      num_C4++;
+      NodeIdx k = node->GetGLoc();
+      std::map<NodeIdx, double>::iterator lb = m_C4Nodes.lower_bound(k);
+      
+      if(lb != m_C4Nodes.end() && !(m_C4Nodes.key_comp()(k, lb->first))) {
```

Comment:
key_comp() will be std::less() which doesn't seem like what you want here.  Is this right?